subworkflows/nf-core/bam_vcf_impute_glimpse2/main.nf

```diff
@@ -0,0 +1,123 @@
+include { GLIMPSE2_CHUNK                     } from '../../../modules/nf-core/glimpse2/chunk/main'
+include { GLIMPSE2_SPLITREFERENCE            } from '../../../modules/nf-core/glimpse2/splitreference/main'
+include { GLIMPSE2_PHASE                     } from '../../../modules/nf-core/glimpse2/phase/main'
+include { GLIMPSE2_LIGATE                    } from '../../../modules/nf-core/glimpse2/ligate/main'
+include { BCFTOOLS_INDEX as BCFTOOLS_INDEX_1 } from '../../../modules/nf-core/bcftools/index/main.nf'
+include { BCFTOOLS_INDEX as BCFTOOLS_INDEX_2 } from '../../../modules/nf-core/bcftools/index/main.nf'
+
+workflow BAM_VCF_IMPUTE_GLIMPSE2 {
+
+    take:
+    ch_input       // channel (mandatory): [ meta, vcf, csi, list, infos ]
+    ch_ref         // channel (mandatory): [ meta, vcf, csi, region ]
+    ch_chunks      // channel (optional) : [ meta, regionin, regionout ]
+    ch_map         // channel (optional) : [ meta, map ]
+    ch_fasta       // channel (optional) : [ meta, fasta, index ]
+    chunk          // val (optional)     : boolean to activate/deactivate chunking step
+    chunk_model    // val (optional)     : model file for chunking
+    splitreference // val (optional)     : boolean to activate/deactivate split reference step
+
+    main:
+
+    ch_versions = channel.empty()
+
+    if ( chunk == true ){
+        // Error if pre-defined chunks are provided when chunking is activated
+        ch_chunks
+            .filter { _meta, regionin, regionout -> regionin.size() > 0 || regionout.size() > 0 }
+            .subscribe {
```

Comment:
Again, this subscribe feels strange to me, and I don't see it used in other subworkflows. Could we use the same ifEmpty pattern?

Reply:
I've changed it to 

```
ch_chunks
            .filter { _meta, regionin, regionout -> regionin.size() == 0 && regionout.size() == 0 }
            .ifEmpty {
                error "ERROR: Cannot provide pre-defined chunks (regionin) when chunk=true. Please either set chunk=false to use provided chunks, or remove input chunks to enable automatic chunking."
            }
```